src/Compilers/CSharp/Portable/Symbols/NamedTypeSymbol.cs

```diff
@@ -713,15 +716,16 @@ internal override bool Equals(TypeSymbol t2, bool ignoreCustomModifiersAndArrayS
 
             // The checks above are supposed to handle the vast majority of cases.
             // More complicated cases are handled in a special helper to make the common case scenario simple/fast
```

Comment:
This is not part of this PR but this comment is confusing. How does the extra helper affect the perf?